### PR TITLE
Update list of oldest test dependencies

### PR DESCRIPTION
**Pull request by dstansby**
I noticed whilst reviewing https://github.com/sunpy/sunpy/pull/4758 that we don't pin everything we should in the oldesttestdeps test run. This adds some extra packages that should be pinned here.

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
For 2.0, will need to drop the hd5 dep.

File: setup.cfg [hunk 1]
```diff
@@ -127,6 +127,9 @@ filterwarnings =
     # is being ignored
     #
     #
+    # See https://github.com/pandas-dev/pandas/issues/26367; this can be
+    # removed when our minimum pandas version is >=0.25
```
Comment (Cadair) [inline setup.cfg:131]: why isn't it already? wasn't that released like  a million years ago now?

Comment (nabobalis) [reply setup.cfg:131]: Should we deal with our min versions for 2.1? If we haven't already?

Comment (dstansby) [reply setup.cfg:131]: Pandas 0.23 doesn't seem compatible with python 3.9 anyway, so I will bump in this PR.

**Review by nabobalis — APPROVED** [review]

**Pull request merged**